TestHelpers/include/TestFixtures.h

```diff
@@ -2,40 +2,232 @@
 #define TESTFIXTURES_H
 
 #include "gtest/gtest.h"
+#include <juce_audio_processors/juce_audio_processors.h>
+#include <Processors/ProcessorGraph/ProcessorGraph.h>
+#include <Processors/SourceNode/SourceNode.h>
+#include <Audio/AudioComponent.h>
+#include <UI/ControlPanel.h>
 
-#define _USE_MATH_DEFINES
-#include <math.h>
+class ProcessorTester {
+public:
+    ProcessorTester(FakeSourceNodeParams fake_source_node_params) {
+        // Singletons...
+        MessageManager::deleteInstance();
 
-class ProcessorTest : public ::testing::Test {
-protected:
-    ProcessorTest() {
-        sn = std::make_unique<FakeSourceNode>();
-        ac = std::make_unique<StubAccessClass>();
+        // initializes the singleton instance
+        MessageManager::getInstance();
+
+        // Reset all state so no interactions between tests.
+        AccessClass::clearAccessClassStateForTesting();
+
+        //Create LookAndFeel object
+        customLookAndFeel = std::make_unique<CustomLookAndFeel>();
+        LookAndFeel::setDefaultLookAndFeel(customLookAndFeel.get());
+
+        
+        // All of these sets the global state in AccessClass in their constructors
+        audio_component = std::make_unique<AudioComponent>();
+        processor_graph = std::make_unique<ProcessorGraph>(true);
+        control_panel = std::make_unique<ControlPanel>(processor_graph.get(), audio_component.get(), true);
+
+        auto sn_temp = new FakeSourceNode(fake_source_node_params);
+        source_node_id = next_processor_id_++;
+        sn_temp->setNodeId(source_node_id);
+        juce::AudioProcessorGraph::Node* n = processor_graph->addNode(
+            std::move(std::unique_ptr<AudioProcessor>(sn_temp)),
+            juce::AudioProcessorGraph::NodeID(source_node_id));
+
+        // Create the source node, and place it in the graph
+        auto sn = (FakeSourceNode *) n->getProcessor();
+        sn->initialize(false);
+        sn->setDestNode(nullptr);
+
+        control_panel->updateRecordEngineList();
+
+        // Refresh everything
+        processor_graph->updateSettings(sn, false);
+    	
+        control_panel ->updateColors();
+
+	}
+
+    ProcessorTester(DataThreadCreator data_thread_creator) {
```

Comment:
The two constructors are very similar. Can you consolidate the two so that there's less copy/paste and thus less room for the implementations to diverge?